plugin/trino-hive/src/test/java/io/trino/plugin/hive/orc/TestHiveOrcWithBloomFilters.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.orc;
+
+import io.trino.plugin.hive.HiveQueryRunner;
+import io.trino.testing.BaseOrcWithBloomFiltersTest;
+import io.trino.testing.QueryRunner;
+
+import static java.lang.String.format;
+
+public class TestHiveOrcWithBloomFilters
+        extends BaseOrcWithBloomFiltersTest
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return HiveQueryRunner.builder()
+                .addHiveProperty("hive.orc.bloom-filters.enabled", "true")
+                .addHiveProperty("hive.orc.default-bloom-filter-fpp", "0.001")
+                .build();
+    }
+
+    @Override
+    protected String getTableProperties(String bloomFilterColumnName, String bucketingColumnName, String sortByColumnName)
+    {
+        return format(
+                "orc_bloom_filter_columns = ARRAY['%s'], bucketed_by = array['%s'], bucket_count = 1, sorted_by = ARRAY['%s']",
```

Comment:
Why is sorting needed for Hive's tests and is not needed for Iceberg?

In particular, was sorting needed to ensure test is not flaky?
Or it wasn't needed, and should be removed from Hive as well (as a prep commit)?

@raunaqmorarka @Praveen2112 i guess you know?

Reply:
> was sorting needed to ensure test is not flaky

Yes to ensure that the bloom filter works always.

Reply:
Okay, so how come we don't need equivalent of this for Iceberg?
is the test going to be flaky?

Reply:
I tested, here only one Orc file is generated, and the number of records is only 15000 lines. Sorting or not has no effect.

Reply:
Or can I submit it separately after Sorted_by PR submission ?

Reply:
For iceberg, only one Orc file is generated in this place, so the filtering condition will not hit the min and Max metadata information of iceberg. @findepi @Praveen2112

Reply:
trino:tpcds1g> select min(totalprice),max(totalprice) from orders;
 _col0  |   _col1   
--------+-----------
 874.89 | 466001.28 
(1 row)

Query 20220512_064516_00007_3b9nu, FINISHED, 1 node
Splits: 10 total, 10 done (100.00%)
0.41 [15K rows, 93.1KB] [36.4K rows/s, 226KB/s]

trino:tpcds1g> drop table orders;
DROP TABLE
trino:tpcds1g> CREATE TABLE orders WITH (format='ORC',partitioning=ARRAY['bucket(orderstatus,1)']) AS SELECT orderstatus, totalprice FROM tpch.tiny.orders;
CREATE TABLE: 15000 rows

Query 20220512_064541_00009_3b9nu, FINISHED, 1 node
Splits: 26 total, 26 done (100.00%)
0.57 [15K rows, 0B] [26.4K rows/s, 0B/s]

trino:tpcds1g> select min(totalprice),max(totalprice) from orders;
 _col0  |   _col1   
--------+-----------
 874.89 | 466001.28 
(1 row)

Query 20220512_064547_00010_3b9nu, FINISHED, 1 node
Splits: 10 total, 10 done (100.00%)
0.22 [15K rows, 93.1KB] [67.3K rows/s, 418KB/s]

Reply:
Can we add an assertion that only one file is created ? Additionally we can add a TODO to specify `partitioning` once we have that feature in iceberg. (Comment was added in Iceberg tests- next commit)

WDYT @findepi

Reply:
> Can we add an assertion that only one file is created ? Additionally we can add a TODO to specify `partitioning` once we have that feature in iceberg. (Comment was added in Iceberg tests- next commit)
> 
> WDYT @findepi

partitioning=ARRARY['bucket(%s, 1)'] ,It has been guaranteed that only one file will be generated during writing. Is it necessary to make additional restrictions?

Reply:
@Praveen2112 
do you want to have "one file" or "one sorted file"?

Reply:
"one sorted file" is kind of good to have. But it might overfit the test case and since having a single file works here. It shouldn't be a problem.

Reply:
> "one sorted file" is kind of good to have. But it might overfit the test case and since having a single file works here. It shouldn't be a problem.

If so, why don't you remove `sorted_by` from Hive's tests?

Reply:
I think we can remove the `sorted_by` condition - as `sorted_by` is added to ensure we won't hit flakiness.

Reply:
![image](https://user-images.githubusercontent.com/40750002/169249270-f3f9731e-b233-44e0-b32f-e03e2132bcf4.png)
For hive run Test 1,000 times without sorted_by.
![image](https://user-images.githubusercontent.com/40750002/169250510-0a60c9cd-3033-4eb2-9b47-52e42db84981.png)
For  iceberg.
Both run 1,000 times and pass the test without sorted_by option.

Reply:
> I think we can remove the `sorted_by` condition - as `sorted_by` is added to ensure we won't hit flakiness.

@Praveen2112 are you saying that without `sorted_by` the test will be flaky?
**IF so**, we cannot remove it **and** we would need equivalent for Iceberg.

Please be explicit.

Reply:
It is added as an additional check to ensure that we don't hit flakkiness in additional to a single file writes - as bloom filters are a probabilistic data structures. But even without this `sorted_by` condition this should work as expected without flakiness as we write the data in a single file. 

Having a control over the order in which the data is written ensures that we encounter a similar bloom filter structure at all times. (But we don't require this - as we check if the filter satisfies or not instead of the content of the bloom filter).

Reply:
> It is added as an additional check to ensure that we don't hit flakkiness in additional to a single file writes - as bloom filters are a probabilistic data structures. 

It sounds like you're saying that this "sorting is needed" ...

> But even without this `sorted_by` condition this should work as expected without flakiness as we write the data in a single file.

... and also that "sorting is not needed".

So is it needed or is it not?

Reply:
> It sounds like you're saying that this "sorting is needed" ...

It more like a additional check - which is not mandatory. 

> So is it needed or is it not?

It is **not needed**...but kind of good to have in future.

Reply:
yeah !! I agree